### PR TITLE
Remove __nonzero__, __long__, __div__, and __rdiv__

### PR DESCRIPTION
**Pull request by eric-wieser**
None of these are real magic methods in Python 3

#### References to other Issues or PRs

Relates to #5916

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/matrices/common.py [hunk 1]
```diff
@@ -2845,7 +2845,7 @@ def __sub__(self, a):
 
     @call_highest_priority('__rtruediv__')
     def __truediv__(self, other):
-        return self.__div__(other)
+        return self.__truediv__(other)
```
Comment (oscarbenjamin) [inline sympy/matrices/common.py:2848]: Is this not an infinite recursion?

File: sympy/matrices/expressions/matexpr.py [hunk 2]
```diff
@@ -135,18 +135,18 @@ def __rpow__(self, other):
...
-    def __rdiv__(self, other):
+    @call_highest_priority('__truediv__')
+    def __rtruediv__(self, other):
         raise NotImplementedError()
         #return MatMul(other, Pow(self, S.NegativeOne))
 
-    __truediv__ = __div__  # type: Callable[[MatrixExpr, Any], Any]
-    __rtruediv__ = __rdiv__  # type: Callable[[MatrixExpr, Any], Any]
+    __truediv__ = __truediv__  # type: Callable[[MatrixExpr, Any], Any]
+    __rtruediv__ = __rtruediv__  # type: Callable[[MatrixExpr, Any], Any]
```
Comment (oscarbenjamin) [inline sympy/matrices/expressions/matexpr.py:149]: I guess this can just be deleted

File: sympy/matrices/tests/test_matrices.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def test_args():
 
 def test_division():
     v = Matrix(1, 2, [x, y])
-    assert v.__div__(z) == Matrix(1, 2, [x/z, y/z])
+    assert v.__truediv__(z) == Matrix(1, 2, [x/z, y/z])
     assert v.__truediv__(z) == Matrix(1, 2, [x/z, y/z])
```
Comment (oscarbenjamin) [inline sympy/matrices/tests/test_matrices.py:49]: I guess this can be deleted as well

File: sympy/polys/rings.py [hunk 4]
```diff
@@ -1349,8 +1349,8 @@ def __truediv__(p1, p2):
     def __rtruediv__(p1, p2):
         return NotImplemented
 
-    __floordiv__ = __div__ = __truediv__
-    __rfloordiv__ = __rdiv__ = __rtruediv__
+    __floordiv__ = __truediv__ = __truediv__
+    __rfloordiv__ = __rtruediv__ = __rtruediv__
```
Comment (oscarbenjamin) [inline sympy/polys/rings.py:1353]: The middle assignment can be removed

File: sympy/matrices/common.py (hunk 1 above)
Comment (eric-wieser) [reply sympy/matrices/common.py:2848]: Yes, this looks like an error. The second commit was done with a find and replace, was waiting for CI to smoke out the bugs

File: sympy/matrices/expressions/matexpr.py (hunk 2 above)
Comment (eric-wieser) [reply sympy/matrices/expressions/matexpr.py:149]: My regex missed this due to the trailing comments, good catch.

**Comment by oscarbenjamin** [comment]
Nice clean up. Thanks

**Pull request merged**